experimenter/experimenter/nimbus-ui/src/components/Summary/index.tsx

```diff
@@ -38,6 +44,51 @@ const Summary = ({ experiment, refetch }: SummaryProps) => {
   const takeawaysProps = useTakeaways(experiment, refetch);
   const qaProps = useQA(experiment, refetch);
   const status = getStatus(experiment);
+  const [legalSignoff, setLegalSignoff] = useState(
+    experiment.legalSignoff || false,
+  );
+  const [qaSignoff, setQaSignoff] = useState(experiment.qaSignoff || false);
+  const [vpSignoff, setVpSignoff] = useState(experiment.vpSignoff || false);
+  const [updateExperiment] = useMutation<
+    updateExperiment,
+    updateExperimentVariables
+  >(UPDATE_EXPERIMENT_MUTATION);
+
+  const handleLegalSignoffChange = (value: boolean) => {
+    setLegalSignoff(value);
+    updateExperimentSignoff(value, qaSignoff, vpSignoff);
+  };
```

Comment:
For these check handler functions, would it be better to only update the local state if the `updateExperimentSignoff` succeeds? You could return a success boolean from `updateExperimentSignoff` or move the try/catch into these functions instead.

Reply:
yup totally agree

Reply:
I tried doing that it and it leads to lot of error, so I gave up for now, but I am adding follow up ticket and can circle back later